File: Clientside/Shop/src/models/ProductDto.ts
import BaseProductDto from "./BaseProductDto";
import BaseVariantDto from "./BaseVariantDto";
import { SizeType } from "./Enums";
import GenderDto from "./GenderDto";
import SearchCategoryDto from "./SearchCategoryDto";

interface ProductDto extends BaseProductDto {
    description: string
    genderId: number
    gender: GenderDto
    sizeType: SizeType
    variants?: BaseVariantDto[]
    categories: SearchCategoryDto[]
}

export default ProductDto;

File: Serverside/Shop/Api/Dtos/Product/ProductDto.cs
using FNO.DataAccess.Database.Models;
using FNO.Shop.ShopApi.Dtos.Variant;
using System.Collections.Generic;

namespace FNO.Shop.ShopApi.Dtos.Product
{
    public class ProductDto : BaseProductDto
    {
        public string Description { get; set; }
        public int GenderId { get; set; }
        public GenderDto Gender { get; set; }
        public Enums.SizeType SizeType { get; set; }
        public List<BaseVariantDto> Variants { get; set; }
        public List<CategoryDto> Categories { get; set; }
    }
}
